src/core/agent/services/keriaNotificationService.ts

```diff
@@ -294,6 +340,55 @@ class KeriaNotificationService extends AgentService {
     }
   }
 
+  async retryFailedNotifications() {
+    const failedNotifications = await this.basicStorage.findById(
+      MiscRecordId.FAILED_NOTIFICATION
+    );
+
+    if (!failedNotifications || !failedNotifications.content) return;
+
+    const { notificationId, attempts, lastAttempt } =
+      failedNotifications.content as unknown as NotificationAttempts;
+    const now = Date.now();
+
+    const backoffDelays = [5000, 10000, 30000, 60000, 300000];
+    const delay =
+      backoffDelays[Math.min(attempts - 1, backoffDelays.length - 1)];
+
+    if (now - lastAttempt >= delay) {
+      try {
+        // Retry process the notification
+        const failedNotifications = await this.basicStorage
+          .findById(MiscRecordId.FAILED_NOTIFICATION)
+          .then((record) => record?.content?.notifications || {});
+
+        for (const notificationId in failedNotifications as Record<
+          string,
+          Notification
+        >) {
+          const failedNotification =
+            failedNotifications[
+              notificationId as keyof typeof failedNotifications
+            ];
+          await this.processNotification(failedNotification);
+        }
+
+        await this.basicStorage.deleteById(notificationId);
+      } catch (error) {
+        await this.basicStorage.createOrUpdateBasicRecord(
```

Comment:
You have to remove the failed notification once you finish processing it. Otherwise we will be re-processing it forever.